crev-lib/src/local.rs

```diff
@@ -985,18 +1033,43 @@ impl ProofStore for Local {
 
     fn proofs_iter(&self) -> Result<Box<dyn Iterator<Item = proof::Proof>>> {
         Ok(Box::new(
-            self.get_proofs_dir_path_opt()?
-                .into_iter()
-                .flat_map(proofs_iter_for_path),
+            self.proofs_for_current_url()?.map(|(proof, _)| proof)
         ))
     }
 }
 
-fn proofs_iter_for_path(path: PathBuf) -> impl Iterator<Item = proof::Proof> {
+/// Scan a directory of git checkouts. Assumes fetch source is the origin URL.
+fn proofs_iter_for_remotes_checkouts(path: PathBuf) -> Result<impl Iterator<Item = (proof::Proof, FetchSource)>>{
+    let dir = std::fs::read_dir(&path)?;
+    Ok(dir
+        .filter_map(|e| e.ok())
+        .filter_map(|e| {
+            if let Ok(ty) = e.file_type() {
+                if ty.is_dir() {
+                    return Some(e.path());
+                }
+            }
+            None
+        })
+        .filter_map(move |path| {
+            let repo = git2::Repository::open(&path).ok()?;
+            let origin = repo.find_remote("origin").ok()?;
+            let fetch_source = FetchSource::Url(Arc::new(Url::new_git(origin.url()?)));
+            Some(proofs_iter_for_path(path, fetch_source))
+        })
+        .flat_map(|iter| iter)
+    )
+}
+
+/// Scan a git checkout or any subdirectory obtained from a known URL
+fn proofs_iter_for_path(path: PathBuf, fetch_source: FetchSource) -> impl Iterator<Item = (proof::Proof, FetchSource)> {
```

Comment:
Seems to me that the caller could just `.map(|p| (p, fetch_source))`, and it would keep this function not being obfuscated about what `fetch_source` does as an argument.

Reply:
Changed